pkg/gossip/resolver/resolver.go

```diff
@@ -38,6 +44,37 @@ func NewResolver(address string) (Resolver, error) {
 	return &socketResolver{typ: "tcp", addr: address}, nil
 }
 
+// SRV returns a slice of addresses from SRV record lookup
+func SRV(name string) ([]string, error) {
+	// Ignore port
+	name, _, err := netutil.SplitHostPort(name, base.DefaultPort)
+	if err != nil {
+		return nil, err
+	}
+
+	if name == "" {
+		// nolint:returnerrcheck
+		return nil, nil
+	}
+
+	// "" as the addr and proto forces the direct look up of the name
+	_, recs, err := lookupSRV("", "", name)
```

Comment:
We have an IANA-assigned port number, so ideally we'd use it here: `_cockroach._tcp.$NAME`. Is that compatible with how k8s sets up its SRV records? If not, this suggests that we may want more options than just SRV or not-SRV (or I guess we could just ask users to pass in the `_cockroach._tcp` bit explicitly if they're using SRV records, but that's ugly and not really how SRV was designed to be used).

Reply:
we discussed that - the point was to be able to transparently "upgrade" an existing cluster deployment to use SRV records without changing the orchestration values for the `--join` flags